### PR TITLE
build tests for hostdb. Add new benchmark for hostdb

### PR DESCRIPTION
**Pull request by cmcfarlen**

### CODE REVIEW

**Comment by cmcfarlen** [comment]
[approve ci ubuntu]

File: src/iocore/hostdb/benchmark_HostDB.cc [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+  RecProcessInit(diags());
+  LibRecordsConfigInit();
+  ink_event_system_init(ts::ModuleVersion(1, 0, ts::ModuleVersion::PRIVATE));
+  ink_net_init(ts::ModuleVersion(1, 0, ts::ModuleVersion::PRIVATE));
+  ink_hostdb_init(HOSTDB_MODULE_PUBLIC_VERSION);
+  ink_dns_init(HOSTDB_MODULE_PUBLIC_VERSION);
+
+  netProcessor.init();
+
+  int nproc = ink_number_of_processors();
```
Comment (bneradt) [inline src/iocore/hostdb/benchmark_HostDB.cc:149]: const

**Review by bneradt — CHANGES_REQUESTED** [review]
Looks good to me. Just a few suggestions.

File: src/iocore/hostdb/benchmark_HostDB.cc [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+  std::vector<StartDNS *> tests;
+
+  for (auto &t : eventProcessor.active_group_threads(ET_CALL)) {
+    StartDNS *startDns = new StartDNS{hosts, t->id, l};
+    t->schedule_imm(startDns);
+    tests.push_back(startDns);
+  }
+
+  FContinuation killer(stop_events);
+  eventProcessor.schedule_in(&killer, HRTIME_SECONDS(300));
```
[the remark this answers is not in the document]
Comment (cmcfarlen) [reply src/iocore/hostdb/benchmark_HostDB.cc:297]: nah, just to cap things if there was a deadlock or something.  I actually extended this to debug something so I'll put it back.

**Review by bneradt — APPROVED** [review]

**Pull request merged**